compiler/parser.nim

```diff
@@ -1764,21 +1764,43 @@ proc parseSection(p: var TParser, kind: TNodeKind,
   else:
     parMessage(p, errIdentifierExpected, p.tok)
 
-proc parseConstant(p: var TParser): PNode =
-  #| constant = identWithPragma (colon typeDesc)? '=' optInd expr indAndComment
-  result = newNodeP(nkConstDef, p)
-  addSon(result, identWithPragma(p))
-  if p.tok.tokType == tkColon:
+proc parseConstTuple(p: var TParser): PNode =
+  result = newNodeP(nkConstTuple, p)
+  getTok(p)                   # skip '('
+  optInd(p, result)
+  while p.tok.tokType in {tkSymbol, tkAccent}:
+    addSon(result, identWithPragma(p))
+    if p.tok.tokType == tkColon:
+      getTok(p)
+      optInd(p, result)
+      addSon(result, parseTypeDesc(p))
+    if p.tok.tokType != tkComma: break
     getTok(p)
-    optInd(p, result)
-    addSon(result, parseTypeDesc(p))
-  else:
-    addSon(result, p.emptyNode)
+
+  addSon(result, p.emptyNode)
+  eat(p, tkParRi)
   eat(p, tkEquals)
   optInd(p, result)
   addSon(result, parseExpr(p))
   indAndComment(p, result)
 
+proc parseConstant(p: var TParser): PNode =
+  #| constant = identWithPragma (colon typeDesc)? '=' optInd expr indAndComment
```

Comment:
This comment describes the grammar rule and stopped reflecting reality. ;-)

Reply:
Also the grammar needs to be written for constTuple

But since I have no idea about EBNF, I leave this to you 😄